closed/test/jtreg-ext/jtreg/openj9/CoreDumpTimeoutHandler.java

```diff
@@ -0,0 +1,142 @@
+ * (c) Copyright IBM Corp. 2020, 2020 All Rights Reserved
+ * ===========================================================================
+ */
+
+package jtreg.openj9;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+
+import com.sun.javatest.regtest.TimeoutHandler;
+
+/**
+ * This is the OpenJ9 core dump timeout handler. It runs jcmd on the process that
+ * has timed out to request a system dump, as well as running jstack as the default
+ * timeout handler does.
+ */
+public class CoreDumpTimeoutHandler extends TimeoutHandler {
+
+    public CoreDumpTimeoutHandler(PrintWriter log, File outputDir, File testJdk) {
+        super(log, outputDir, testJdk);
+    }
+
+    @Override
+    protected void runActions(Process proc, long pid) throws InterruptedException {
+        runJcmd(pid);
+        runJstack(pid);
+    }
+
+    /**
+     * Run jstack on the specified pid.
+     * @param pid Process Id
+     */
+    private void runJstack(long pid) throws InterruptedException {
+        try {
+            log.println("Running jstack on process " + pid);
+
+            File jstack = findJstack();
+            if (jstack == null) {
+                log.println("Warning: Could not find jstack in: " + testJdk.getAbsolutePath());
+                log.println("Will not dump jstack output.");
+                return;
+            }
+
+            ProcessBuilder pb = new ProcessBuilder(jstack.getAbsolutePath(), pid + "");
+            pb.redirectErrorStream(true);
+
+            Process p = pb.start();
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    log.println(line);
+                }
+                p.waitFor();
+            }
+        } catch (IOException ex) {
+            ex.printStackTrace(log);
+        }
+    }
+
+    private File findJstack() {
+        File jstack = new File(new File(testJdk, "bin"), "jstack");
+        if (!jstack.exists()) {
+            jstack = new File(new File(testJdk, "bin"), "jstack.exe");
+            if (!jstack.exists()) {
+                return null;
+            }
+        }
+        return jstack;
+    }
+
+    /**
+     * Run jcmd on the specified pid.
+     * @param pid Process Id
+     */
+    private void runJcmd(long pid) throws InterruptedException {
+        try {
+            log.println("Running jcmd on process " + pid);
+
+            File jcmd = findJcmd();
+            if (jcmd == null) {
+                log.println("Warning: Could not find jcmd in: " + testJdk.getAbsolutePath());
+                log.println("Will not run jcmd.");
+                return;
+            }
+
+            ProcessBuilder pb = new ProcessBuilder(jcmd.getAbsolutePath(), pid + "", "Dump.system");
```

Comment:
Suggest `Long.toString(pid)` would be clearer than `pid + ""`.

Reply:
Updated here, but I didn't change the original code for jstack.